portal-api/src/v1/validation/fields/cover-dates.js

```diff
@@ -1,49 +1,45 @@
-const moment = require('moment');
+const { isValid, set, startOfDay, isBefore, isSameDay } = require('date-fns');
 const { orderNumber } = require('../../../utils/error-list-order-number');
-const {
-  dateHasAllValues,
-} = require('./date');
-const { formattedTimestamp } = require('../../facility-dates/timestamp');
 const isReadyForValidation = require('../helpers/isReadyForValidation.helper');
+const { hasValue } = require('../../../utils/string');
 
 module.exports = (submittedValues, deal, errorList) => {
   const newErrorList = errorList;
 
-  const requestedCoverStartDateTimestamp = formattedTimestamp(submittedValues.requestedCoverStartDate);
-
   const {
     'coverEndDate-day': coverEndDateDay,
     'coverEndDate-month': coverEndDateMonth,
     'coverEndDate-year': coverEndDateYear,
   } = submittedValues;
+  const requestedCoverStartDate = new Date(Number.parseInt(submittedValues.requestedCoverStartDate, 10));
 
   if (isReadyForValidation(deal, submittedValues)) {
-    const hasValidRequestedCoverStartDate = (requestedCoverStartDateTimestamp && !newErrorList.requestedCoverStartDate);
-
-    const hasValidCoverEndDate = dateHasAllValues(
-      coverEndDateDay,
-      coverEndDateMonth,
-      coverEndDateYear,
-    ) && !newErrorList.coverEndDate;
-
-    const hasValidCoverStartAndEndDates = (hasValidRequestedCoverStartDate && hasValidCoverEndDate);
-
-    if (hasValidCoverStartAndEndDates) {
-      const requestedCoverStartDateDay = moment(requestedCoverStartDateTimestamp).format('DD');
-      const requestedCoverStartDateMonth = moment(requestedCoverStartDateTimestamp).format('MM');
-      const requestedCoverStartDateYear = moment(requestedCoverStartDateTimestamp).format('YYYY');
-
-      const requestedCoverStartDate = `${requestedCoverStartDateYear}-${requestedCoverStartDateMonth}-${requestedCoverStartDateDay}`;
-      const coverEndDate = `${coverEndDateYear}-${coverEndDateMonth}-${coverEndDateDay}`;
-
-      if (moment(coverEndDate).isBefore(requestedCoverStartDate)) {
+    const hasValidRequestedCoverStartDate =
+      submittedValues.requestedCoverStartDate &&
+      isValid(requestedCoverStartDate) &&
+      !newErrorList.requestedCoverStartDate;
+
+    const hasValidCoverEndDate =
+      hasValue(coverEndDateDay) &&
+      hasValue(coverEndDateMonth) &&
+      hasValue(coverEndDateYear) &&
+      !newErrorList.coverEndDate;
+
+    if (hasValidRequestedCoverStartDate && hasValidCoverEndDate) {
+      const coverEndDate = set(startOfDay(new Date()), {
```

Comment:
As per `supply-contract-conversion-rate-date-is-required-for-non-GBP.js` (what a mouthful!), if we're only caring about dates, lets work as `startOfDay` throughout please for simplicity -- what do you think?

Also same comments on this file as `supply-contract-conversion-rate-date-is-required-for-non-GBP.js`